src/metadata.cpp

```diff
@@ -47,6 +47,59 @@ size_t getTextLength(const std::string& utf8EncodedString)
   return icu::UnicodeString::fromUTF8(utf8EncodedString).length();
 }
 
+bool checkTextLanguage(const std::string& text, const std::string& langCode)
+{
+  // TODO: check that text is in langCode's script
+  return true;
+}
+
+class MetadataComplexCheckBase
+{
+public:
+  const std::string description;
+  const MetadataComplexCheckBase* const prev;
+
+public: // functions
+  explicit MetadataComplexCheckBase(const std::string& desc)
+    : description(desc)
+    , prev(last)
+  {
+    last = this;
+  }
+
+  virtual ~MetadataComplexCheckBase() = default;
+
+  virtual bool checkMetadata(const Metadata& m) const = 0;
+
+  static const MetadataComplexCheckBase* getLastCheck() { return last; }
+
+private: // functions
+  static const MetadataComplexCheckBase* last;
```

Comment:
Keeping a static pointer to `this` is risky as the object can be moved. 
We only create global object for now but it is still tricky.
If we do that, it would be better to at least delete copy/move constructor and assignment operator.